plugins/early_talker.js

```diff
@@ -64,3 +76,56 @@ exports.early_talker = function(next, connection) {
 
     setTimeout(function () { check(); }, pause);
 };
+
+
+/**
+ * Check if an ip is whitelisted
+ *
+ * @param  {String} ip       The remote IP to verify
+ * @return {Boolean}         True if is whitelisted
+ */
+exports.ip_in_list = function (ip) {
+    var plugin = this;
+    var ipobj = ipaddr.parse(ip);
```

Comment:
This is a guess, but the connection in this case is from
haraka-test-fixtures, and doesn't implement remote_ip

https://github.com/haraka/haraka-test-fixtures/blob/master/lib/connection.js

On Wed, Apr 27, 2016 at 6:58 PM, Matt Simerson <notifications@github.com>
wrote:

> In plugins/early_talker.js
> <https://github.com/haraka/Haraka/pull/1423#discussion_r61351058>:
>
> > @@ -64,3 +76,56 @@ exports.early_talker = function(next, connection) {
> >
> >      setTimeout(function () { check(); }, pause);
> >  };
> > +
> > +
> > +/**
> > + * Check if an ip is whitelisted
> > + *
> > + * @param  {String} ip       The remote IP to verify
> > + * @return {Boolean}         True if is whitelisted
> > + */
> > +exports.ip_in_list = function (ip) {
> > +    var plugin = this;
> > +    var ipobj = ipaddr.parse(ip);
>
> Why parse the IP if plugin.whitelist doesn't exist? (shouldn't that check
> come before this one?)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/haraka/Haraka/pull/1423/files/75985027341d4624c91d48dcf0438495caba25d1#r61351058>
>